src/Message/Message.php

```diff
@@ -37,12 +37,4 @@ public function getMetadata(): array
     {
         return $this->metadata;
     }
-
-    public function withMetadata(array $metadata): self
```

Comment:
why?

Reply:
What's the real-world use case? I don't see any, that's why I removed this method.

Reply:
Message handlers may add metadata into the message